e2e/etcdctlv3_test.go

```diff
@@ -100,6 +100,57 @@ func testCtlV3Set(t *testing.T, cfg *etcdProcessClusterConfig, dialTimeout time.
 	}
 }
 
+func TestCtlV3Watch(t *testing.T) {
+	testCtlV3Watch(t, &configNoTLS, 5*time.Second, false, false)
+}
+
+func TestCtlV3WatchInteractive(t *testing.T) {
+	testCtlV3Watch(t, &configNoTLS, 5*time.Second, false, true)
+}
+
+func testCtlV3Watch(t *testing.T, cfg *etcdProcessClusterConfig, dialTimeout time.Duration, quorum, interactive bool) {
+	defer testutil.AfterTest(t)
+
+	os.Setenv("ETCDCTL_API", "3")
+	epc := setupCtlV3Test(t, cfg, quorum)
+	defer func() {
+		os.Unsetenv("ETCDCTL_API")
+		if errC := epc.Close(); errC != nil {
+			t.Fatalf("error closing etcd processes (%v)", errC)
+		}
+	}()
+
+	key, value := "foo", "bar"
+
+	errc := make(chan error, 1)
+	expectTimeout := dialTimeout > 0 && dialTimeout <= time.Nanosecond
+	go func() {
+		defer close(errc)
+		if err := ctlV3Watch(epc, key, value, dialTimeout, interactive); err != nil {
+			if expectTimeout && isGRPCTimedout(err) {
+				errc <- fmt.Errorf("watch error (%v)", err)
+				return
+			}
+		}
+	}()
+
+	time.Sleep(2 * time.Second) // wait for watch to start
```

Comment:
the test shouldn't need to sleep if the etcdctl watch is set to start at revision 1